### PR TITLE
Fix content types fields dependency

### LINKED ISSUE #12488: When using ContentTypes without any fields, the "Fields" section should not show up
### Describe the bug
When we enable `OC.ContentTypes` feature but not enable any feature that compose fields like `OC.ContentFields`, the Fields section shows up.

image

### To Reproduce
1. Enable OC.ContentTypes and disable OC.ContentFields.
2. Go to Content Types and edit or add a new content type

### Suggestion
I think we should remove the "Add field" button when there is no fields definitions. Also, if a currently attached field has no definition, we should not be able to edit it, but we should be able to remove it. This can happen if a field was attach to a content type/part while the field definition is available but then the feature that provides the definition was disabled. 

Moreover, we should remove the "Fields" section from the UI if the current content-type has no fields and also there is no field definitions.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #12488

The button shows up when at least 1 field definition is available. 
image

A field was previously added to the content items. but there are no more field definition registred, so we allow removing previously attached fields but there is no Add Fields button

image

There is no content fields available, so we hide the button
image

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@jtkech are you okay with me merging this PR?

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs [hunk 1]
```diff
@@ -595,7 +604,7 @@ public async Task<ActionResult> AddFieldTo(string id, string returnUrl = null)
             var viewModel = new AddFieldViewModel
             {
                 Part = partViewModel.PartDefinition,
-                Fields = _contentDefinitionService.GetFields().Select(x => x.Name).OrderBy(x => x).ToList()
+                Fields = fields.Select(x => x.Name).OrderBy(x => x).ToList()
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs:607]:
```suggestion
                Fields = fields.Select(f => f.Name).OrderBy(f => f).ToList()
```

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs [hunk 2]
```diff
@@ -617,10 +626,17 @@ public async Task<ActionResult> AddFieldToPOST(AddFieldViewModel viewModel, stri
                 return NotFound();
             }
 
+            var fields = _contentDefinitionService.GetFields().ToList();
+
+            if (!fields.Any(x => x.Name.Equals(viewModel.FieldTypeName, StringComparison.OrdinalIgnoreCase)))
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs:631]:
```suggestion
            if (!fields.Any(f => f.Name.Equals(viewModel.FieldTypeName, StringComparison.OrdinalIgnoreCase)))
```

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs [hunk 3]
```diff
@@ -693,7 +709,8 @@ public async Task<ActionResult> EditField(string id, string name, string returnU
 
             var partFieldDefinition = partViewModel.PartDefinition.Fields.FirstOrDefault(x => String.Equals(x.Name, name, StringComparison.OrdinalIgnoreCase));
 
-            if (partFieldDefinition == null)
+            if (partFieldDefinition?.FieldDefinition?.Name == null
+                || !_contentDefinitionService.GetFields().Any(x => x.Name.Equals(partFieldDefinition.FieldDefinition.Name, StringComparison.OrdinalIgnoreCase)))
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs:713]:
```suggestion
                || !_contentDefinitionService.GetFields().Any(f => f.Name.Equals(partFieldDefinition.FieldDefinition.Name, StringComparison.OrdinalIgnoreCase)))
```

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/Views/Admin/Edit.cshtml [hunk 4]
```diff
@@ -38,40 +42,47 @@
...
-            @if (typePart != null)
-            {
+            <ul class="list-group sortable" id="fields">
                 @foreach (var field in orderedFields)
                 {
                     <li class="list-group-item" style="cursor: move;">
                         <div class="w-100">
                             <div class="float-end">
-                                <a asp-route-action="EditField" asp-route-id="@Model.TypeDefinition.Name" asp-route-name="@field.Name" asp-route-returnUrl="@FullRequestPath" class="btn btn-primary btn-sm" role="button">@T["Edit"]</a>
+                                @if (fields.Any(x => x.Name.Equals(field.FieldDefinition.Name, StringComparison.OrdinalIgnoreCase)))
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/Views/Admin/Edit.cshtml:56]:
```suggestion
                                @if (fields.Any(f => f.Name.Equals(field.FieldDefinition.Name, StringComparison.OrdinalIgnoreCase)))
```

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/Views/Admin/Edit.cshtml [hunk 5]
```diff
@@ -16,10 +17,13 @@
         .OrderBy(x => x.Order)
         .Select(x => x.Part);
 
-    var orderedFields = typePart == null ? Enumerable.Empty<ContentPartFieldDefinition>() : typePart.PartDefinition.Fields
+    var orderedFields = ((typePart?.PartDefinition?.Fields) ?? Enumerable.Empty<ContentPartFieldDefinition>())
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/Views/Admin/Edit.cshtml:20]: Why you didn't assign `Enumerable.Empty<ContentPartFieldDefinition>()` to `orderedFields` if the fields is `null`

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by hishamco — APPROVED** [review]
LGTM, could you please provide a screenshots?

**Comment by MikeAlhayek** [comment]
@hishamco I updated the PR with screenshots. I am checking out a different PR, if you want to do more testing, please fork this PR and test it out or evaluate it locally. Not planning any additional changes, unless someone spots a bug that I missed.

**Review by jtkech — APPROVED** [review]

**Pull request merged**